tests/virt_autotest/kubevirt_tests_agent.pm

```diff
@@ -43,15 +46,22 @@ sub rke2_agent_setup {
     my ($self, $server_ip) = @_;
 
     record_info('Start RKE2 agent setup', '');
-    systemctl('stop apparmor.service');
-    systemctl('stop firewalld.service');
+    unless (is_transactional) {
+        disable_and_stop_service('apparmor.service');
+        disable_and_stop_service('firewalld.service');
+    }
     $self->setup_passwordless_ssh_login($server_ip);
 
-    # Ensure SUSE certificates are installed on the node
+    # rebootmgr has to be turned off as prerequisity for this to work
+    script_run "rebootmgrctl set-strategy off" if (is_transactional);
+    # Check if the package 'ca-certificates-suse' are installed on the node
     ensure_ca_certificates_suse_installed();
+    transactional::process_reboot(trigger => 1) if (is_transactional);
 
-    # Install downstream kubevirt packages
-    zypper_call('in -n kubernetes1.18-client') if (script_run('rpmquery kubernetes1.18-client'));
+    unless (is_transactional) {
+        # Install downstream kubevirt packages
+        zypper_call('in -n kubernetes1.18-client') if (script_run('rpmquery kubernetes1.18-client'));
+    }
 
```

Comment:
I think you can group your codes into "transactional" and "non-transactional" blocks.

Reply:
I would like to remove this part. The package is not required currently, as the kubectl command is provided by rke2 installation.

Reply:
I think you can organize them better if possible. Now it seems small code blocks litter everywhere.

Reply:
The snippet of code is useless and removed, please review again. Thanks